applications/StructuralMechanicsApplication/python_scripts/adjoint_structural_solver.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import structural_mechanics_solver
+
+# Check that KratosMultiphysics was imported in the main script
+KratosMultiphysics.CheckForPreviousImport()
+
+def CreateSolver(main_model_part, custom_settings):
+    return AdjointStructuralSolver(main_model_part, custom_settings)
+
+class AdjointStructuralSolver(structural_mechanics_solver.MechanicalSolver):
+
+    def __init__(self, main_model_part, custom_settings):
+
+        adjoint_settings = KratosMultiphysics.Parameters("""
+        {
+            "scheme_settings" : {
+                "scheme_type": "structural"
```

Comment:
This is not very descriptive, I would use use sth like `adjoint_structural` (same as what you use for the scheme)